src/linear_combination.jl

```diff
@@ -0,0 +1,77 @@
+"""
+    LinearCombination(coefficients) <: Transform
+
+Calculate the linear combination using the column weights passed in.
+"""
+struct LinearCombination <: Transform
+    coefficients::Vector{Real}
+end
+
+function _check_dimensions_match(LC::LinearCombination, num_inds)
+    num_coefficients = length(LC.coefficients)
+    if num_inds != num_coefficients
+        throw(DimensionMismatch(
+            "Size $num_inds doesn't match number of coefficients ($num_coefficients)"
+        ))
+    end
+end
+
+_sum_row(row, coefficients) = sum(map(*, row, coefficients))
+
+"""
+    apply(x::AbstractVector, LC::LinearCombination; inds=Colon())
+
+Applies the [`LinearCombination`](@ref) to each of the specified indices in `x`.
+
+If no `inds` are specified, then the [`LinearCombination`](@ref) is applied to all elements.
+"""
+function apply(x::AbstractVector, LC::LinearCombination; inds=Colon())
```

Comment:
I left `dims` out for AbstractVectors as it doesn't make sense for them

Reply:
Can we not just generalize the `AbstractArray` code to handle a one dimensional array?

Reply:
I was running into problems bc Vectors and Matrices work slightly differently, also a Vector  is not an AbstractMatrix which is what we are restricting the other method to now